demos/cubes/src/main.rs

```diff
@@ -342,15 +342,13 @@ fn main() {
             let mut nodes = node_store.write();
             let mut node_maybe = nodes.first();
             while let Some(node_ptr) = node_maybe {
-                let (local, parent) = {
-                    let n = nodes.access(&node_ptr);
-                    (n.local.clone(), n.parent.clone()) //FIXME
+                let local = nodes.access(&node_ptr).local;
+                let parent_world = match nodes.access(&node_ptr).parent {
+                    Some(ref parent) => Some(nodes.access(parent).world),
+                    None => None,
                 };
-                let world = match parent {
-                    Some(parent) => nodes.access(&parent).world.concat(&local),
-                    None => local,
-                };
-                nodes.access(&node_ptr).world = world;
+                let world = parent_world.map_or(local, |p| p.concat(&local));
```

Comment:
What's the reason of making `parent_world` via a `match` just to use `map_or` here right afterwards? Let's have everything computed in the `match` right away, like it used to.